lib/logthrdest/logthrdestdrv.c

```diff
@@ -107,19 +107,15 @@ log_threaded_dest_worker_rewind_messages(LogThreadedDestWorker *self, gint batch
 static gchar *
 _format_queue_persist_name(LogThreadedDestWorker *self)
 {
-  LogPipe *owner = &self->owner->super.super.super;
-
   if (self->worker_index == 0)
     {
       /* the first worker uses the legacy persist name, e.g.  to be able to
        * recover the queue previously used.  */
-      return g_strdup(log_pipe_get_persist_name(owner));
+      return g_strdup(self->owner->persist_name);
```

Comment:
Not particularity this line, but in general this code violates the contract of `log_pipe_get_persist_name` gives me the persist-name. It would be really hard for me to remember that I no longer can use that in case of `LogThreadedDest...`.

Instead, I would keep the new init as you did.
Add a new general `generate_persist_name` to `LogThreadedDestDriver`
```c
static const gchar *
_persist_name(const LogPipe *s)
{
  LogThreadedDestDriver *self = (LogThreadedDestDriver*)s;
  g_assert(self->persist_name);
  return self->persist_name;
}
```
That simply always returns with the cache value (it is cached in the new init anyway), and simply do not override the `generate_persist_name` in the classes inherited from this one. With that here you do not need to change a thing.

Reply:
Thank You for the review! Based on your observations, I asked myself again: Why am I doing this?

> I try to force the generate_persist_name virtual function in the inherited classes.

To be more precise: I try to force the case, when it is safe to call it. (If we call it before certain init steps, it could end up in a null pointer dereference.)

My observations:

- Providing a LogThreadedDestDriver implementation of the generate_persist_name function do not prevent overriding it in the inherited classes. So any safeguard in that function will be futile.
- Overriding the generate_persist_name function is not a bad thing. I can still safely call the log_pipe_get_persist_name with it. Which is much more better than my current implementation.
- I only started to store the provided value to avoid any accidental mismatch between the init/deinit phases. i.e. the drivers usually generate the persist name from connection details (IP, Port, etc..), which might change during the init step.
  

It might be possible to force this stability without breaking the log_pipe_get_persist_name contract. proposal:
 - Keep the new init method, and store the provided value.
 - Do not access the stored value in the code directly, keep the log_pipe_get_persist_name function. (Which implicitly means, we will continue to use the generate_persist_name functions of the inherited classes.)
 - Compare the output of the log_pipe_get_persist_name function with the stored value in the key stages (init_method, on_config_inited, deinit) as a safeguard around the current/future implementations. That will pop out any programming error with the first test runs.

Reply:
Discussed with @Kokan IRL. As it turned out my description was not clear enough. Since the earlier code had the highlighted problems, there were no reason keeping it. I updated the pull request with the proposal, and marking it as WIP while we discuss it.